src/data/repositories/InstanceDefaultRepository.ts

```diff
@@ -1,25 +1,116 @@
 import { InstanceRepository } from "../../domain/repositories/InstanceRepository";
 import { cache } from "../../utils/cache";
 import { getD2APiFromInstance } from "../../utils/d2-api";
 import { apiToFuture } from "../../utils/futures";
+import { DataStoreClient } from "../data-store/DataStoreClient";
+import { DataStoreKeys } from "../data-store/DataStoreKeys";
 import { Instance } from "../entities/Instance";
 
 export class InstanceDefaultRepository implements InstanceRepository {
     private api: D2Api;
 
-    constructor(instance: Instance) {
+    constructor(instance: Instance, private dataStoreClient: DataStoreClient) {
         this.api = getD2APiFromInstance(instance);
     }
 
     public getBaseUrl(): string {
         return this.api.baseUrl;
     }
 
+    mapUserOrgUnitsAccess = (
+        organisationUnits: SelectedPick<
+            D2OrganisationUnitSchema,
+            {
+                id: true;
+                name: true;
+            }
+        >[],
+        dataViewOrganisationUnits: SelectedPick<
+            D2OrganisationUnitSchema,
+            {
+                id: true;
+                name: true;
+            }
+        >[]
+    ): OrgUnitAccess[] => {
+        const orgUnitsAccess = organisationUnits.map(ou => ({
+            orgUnitId: ou.id,
+            orgUnitName: ou.name,
+            readAccess: dataViewOrganisationUnits.findIndex(dvou => dvou.id === ou.id) > -1 ? true : false,
+            captureAccess: true,
+        }));
+
+        //Setting view access for org units that are present in dataViewOrganisationUnits and not organisationUnits
+        dataViewOrganisationUnits.forEach(dvou => {
+            if (orgUnitsAccess.findIndex(ou => ou.orgUnitId === dvou.id) === -1) {
+                orgUnitsAccess.push({
+                    orgUnitId: dvou.id,
+                    orgUnitName: dvou.name,
+                    readAccess: true,
+                    captureAccess: false, //orgUnits in dataViewOrganisationUnits dont have capture access
+                });
+            }
+        });
+
+        //TO DO: TEMP - For testing the OrgUnit Access implementation, consoling the orgUnitAccess.
+        //TO DO: Remove once permission implementation done.
+        console.debug("Org Unit Access Permissions : " + JSON.stringify(orgUnitsAccess));
+        return orgUnitsAccess;
+    };
+
+    mapUserGroupAccess = (
+        userGroups: SelectedPick<
+            D2UserGroupSchema,
+            {
+                id: true;
+                name: true;
+            }
+        >[]
+    ): FutureData<ModuleAccess[]> => {
+        return this.dataStoreClient.listCollection<GlassModule>(DataStoreKeys.MODULES).flatMap(modules => {
+            //Iterate through modules and populate access for each
+            const moduleAccess = modules.map(module => {
+                let readAccess = false;
+                let writeAccess = false;
+                userGroups.forEach(userGroup => {
```

Comment:
[Functional programming] Here you'd use `some`: `const readAccess = _(userGroups).some(...)`

Reply:
Addressed in https://github.com/EyeSeeTea/glass-dev/pull/35/commits/2d6776ef32787b23c18581b271cd4d3670d94b21